mmcv/ops/nms.py

```diff
@@ -218,6 +218,15 @@ def soft_nms(boxes,
                                      float(iou_threshold), float(sigma),
                                      float(min_score), method_dict[method],
                                      int(offset))
+
+    if torch.onnx.is_in_onnx_export():
```

Comment:
It seems that size(0) would be treated as a tensor. There is no need to add this branch.
![tmp onnx](https://user-images.githubusercontent.com/1239736/117237353-608d5a00-ae5d-11eb-8925-98f0e65b4665.png)

Reply:
```python
a = torch.rand((3,4))
b=a.size(0)
c=torch._shape_as_tensor(a)[0]
type(b)  # <class 'int'>
type(c) # <class 'torch.Tensor'> with ndim 0

```

Reply:
`Tensor.size()` and `Tensor.shape` would be treated as [Shape](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Shape) op in ONNX even when their type is `int` in python.
You can visualize the generated ONNX at https://netron.app/.